### PR TITLE
[RFC] Add GB300 scale validation instruction for Perf Maxxing

### PR DESCRIPTION
**Pull request by wwwjn**
## Summary

This RFC adds a staged validation workflow for DeepSeek V3 changes that are too large for regular CI. The goal is to catch correctness, packaging, memory, and communication regressions before spending a full 256-GPU allocation.

The ladder has three layers:

| Layer | Coverage | Gate |
| --- | --- | --- |
| Single node | Existing B200 CI runs a 4-GPU DeepSeek V3 MXFP8/FSDP/PP/EP configuration | Required CI coverage for the core components |
| Multi-node | A 16-GPU GB300 Slurm test runs the **debug model** with PP2, FSDP8, EP8, MXFP8, and loss compile | Must match the checked-in full-precision loss and gradient-norm golden |
| Real scale | A 256-GPU (@wwwjn: need to be update to 512 GPUs) DeepSeek V3 671B test runs the PP4/DP64/EP32 MXFP8 starting configuration | Must be launched separately, only after the 16-GPU result and artifacts are accepted |

## How to Verifying a PR for perf-maxing workflow

1. A**uto-triggered by CI**: Trigger Single-node B200 coverage by labeling the PR with `ciflow/b200` tag.
2. **User-launched**: Run only the 16-GPU gate on GB300 Cluster, following instruction in P2495391449
3. **User-launched**:  Proceed explicitly to 256 GPUs on GB300 Cluster, following instruction in P2495391449

A user could launching in internal GB300 cluster by following P2495391449, which will eventually be shared as a wiki.

## Example Verification Output (need to be included in every PR)

Then the user (or letting an agent) to collect following information to make the run verifiable by collaborators. 
```
  Run identity

  - Suite: gb300_dsv3 / deepseek_v3_mxfp8_pp2_ep8_loss_compile
  - Slurm job ID: 1275318
  - Output directory: 16gpu-validation-01 (torchtitan-gb300-validation/d286780-20260908/16gpu-
    validation-01/deepseek_v3_mxfp8_pp2_ep8_loss_compile)

  Source

  - TorchTitan revision: 72d8720ad1025fa411d8e53e18724d96381f3714
  - Branch: validation/gb300-dsv3-d286780
  - Clean checkout: Yes
  - Tracked diff SHA256: None
  - Note: its Git tree exactly matches pushed commit 913df900c.

  Runtime

  - Python: /storage/home/jianiw/venvs/torchtitan-dsv3-gb300-d286780-py31214-20260908/bin/
    python3, Python 3.12.14

  - PyTorch: 2.15.0.dev20260908+cu130, revision 3d2b4c7639df58a777529a4699efa74b00619d90
  - CUDA / NCCL: CUDA 13.0; NCCL 2.30.7
  - TorchAO: 0.19.0.dev20260908+git3efa9eda3
  - TorchAO wheel SHA256: 8fd81d22f7d92f6e2a7b21e598af9ae898ef8ffba827b0e6c70c1d0e775ea2fc
  - Triton: 3.8.0+gitc01b6774
  - NCCL environment: cluster-provided IB/NVLS settings retained; NCCL_DEBUG=WARN,
    NCCL_IB_HCA=ibp, NCCL_NET_PLUGIN=none, TORCH_NCCL_TRACE_BUFFER_SIZE=2000. Job config
    changed TORCH_NCCL_ASYNC_ERROR_HANDLING from 1 to 3.


  Results

  - Slurm state / exit code / elapsed time: COMPLETED / 0:0 / 00:06:16; Python step 00:05:12
  - Topology packing: Properly packed in one four-node segment, block_DH3-029-US-EAST-04B
  - Golden loss: Exact match on all 10 steps; final 4.073132514953613
  - Golden grad norm: Exact match on all 10 steps; final 1.972544550895691
  - Throughput: 5,250.8–5,297.0 logged TPS during unprofiled steady steps 4–6
  - Peak memory: rank-0 maximum active 0.4835 GiB; reserved 0.6172 GiB; zero allocation retries
    and OOMs
```

## Contents in this RFC
- Implement the test ladders by reusing current torchtitan launchers. 
- Provided an instruction for users to kick-off GB300 verification runs P2495391449. 
- Add recipe config for DSV3 as a starting point, people should add to this centralized config while landing their PRs.
- `loss_compare.py` can use a synchronous launcher command, allowing the same deterministic seed-42 loss/gradient-norm workflow to submit through Slurm.
- `scripts/collect_environment.py` records source identity, the live compute runtime, installed distributions, PyTorch/CUDA/NCCL versions, selected NCCL variables, local GPU topology, and Slurm allocation facts.
    - Environment collection is opt-in. The scale runner enables it, and`multinode_trainer.slurm` executes it on a compute node with the same `python3` used for distributed training.
    - The compute run retains `environment.json`, `resolved_config.json`, the full training log, TensorBoard metrics, profiler traces, and memory snapshots. Trace/memory interpretation and the PR report intentionally remain outsid the compute runner.

## Review requested

As an RFC, feedback is especially welcome on:
- whether the recipe set is sufficient for reproducible scale claims and it the config good as a starting point of ;
- the ladder of tests: single node B200 CI ->  16-GPU on GB300 -> explicit approval -> 256-GPU on GB300;
- the generic instruction to verify the PR P2495391449

### CODE REVIEW

File: torchtitan/models/deepseek_v3/config_registry.py [hunk 1]
```diff
@@ -300,3 +300,82 @@ def deepseek_v3_671b_float8(seq_len: int | None = None) -> Trainer.Config:
         ],
     )
     return config
+
+
+# -----------------------------------------------------------------------------
+# Perf maxxing
```
Comment (wwwjn) [inline torchtitan/models/deepseek_v3/config_registry.py]: Not sure if the following configs are a good starting point or not. Need your input for the starting point config (or integrating what we already landed today in torchtitan)

File: tests/assets/losses/gb300/deepseek_v3_mxfp8_16gpu.txt
Comment (tianyu-l) [inline tests/assets/losses/gb300/deepseek_v3_mxfp8_16gpu.txt:1]: sounds cluster specific, and the cluster is not owned by titan repo's CI -- may be good to clarify the setup

File: multinode_trainer.slurm
Comment (tianyu-l) [inline multinode_trainer.slurm:1]: I wonder if it make sense to put this in a `launcher` folder. Benefit is that
- later people may put more scripts in it
- we can put collect_environment.py in it, instead of dumping unrelated things together all in `scripts/`

**Review by tianyu-l — COMMENTED** [review]
makes sense to me in general, awaiting @sanketpurandare input
